Mathlib/Topology/Algebra/ProperAction.lean

```diff
@@ -0,0 +1,342 @@
+      change IsProperMap (Prod.map ((↑) : H → G) (fun x ↦ x))
+      exact IsProperMap.prod_map (isProperMap_subtype_val_of_closed H_closed) isProperMap_id
+    have : IsProperMap (fun hx : H × X ↦ (hx.1 • hx.2, hx.2)) := by
+      change IsProperMap ((fun gx ↦ (gx.1 • gx.2, gx.2)) ∘
+        (fun hx : H × X ↦ ((hx.1, hx.2) : G × X)))
+      exact this.comp (isProperMap_smul_pair G X)
+    assumption
+
+/-
+Some suggestions of things to prove,
+taken from Kapovich
+-/
+
+/-
+If `X` is T2 and first countable,
+then the naive definition
+of proper action is equivalent to the good definition
+(Kapovich Th 11)
+-/
+theorem naiveProper_iff_ProperSMul_T2_FirstCountable
+    [T2Space X] [FirstCountableTopology X] :
+    ProperlyDiscontinuousSMul G X ↔ ProperSMul G X
+    := by sorry
+
+lemma tendsTo_comp_continuous
+    {lx: Filter X} {f : X → Y} {g : Y → Z} {y : Y}
+    (H : Tendsto f lx (𝓝 y)) (hg: Continuous g) :
+    Tendsto (g ∘ f) lx (𝓝 (g y)) := by
+  apply Filter.Tendsto.comp _ H
+  exact hg.tendsto y
+
+/-
+If `X` and `Y` are T2 and first countable,
+then the naive definition
+of proper map is equivalent to the good definition
+-/
+theorem properMap_of_naiveProper_T2_FirstCountable
+    [T2Space X] [FirstCountableTopology X]
+    [T2Space Y] [FirstCountableTopology Y]
+    (f : X → Y) (hcont: Continuous f):
+    (∀ (K : Set Y), (IsCompact K → IsCompact (f ⁻¹' K)))
+    → IsProperMap f
+    := by
+  intro h
+  -- a map is proper iff it is closed and the fibers are compacts
+  rw [isProperMap_iff_isClosedMap_and_compact_fibers]
+  refine ⟨(by assumption), ?_, ?_ ⟩
+  · rw [IsClosedMap]
+    intro U closed_U
+    -- in a first countable space, a set is closed iff sequentially closed
+    apply IsSeqClosed.isClosed
+    -- introduce a converging sequence in the image f(U)
+    intro seq y seq_in_image seq_conv
+    -- pullback this sequence to a sequence in U
+    choose s' hs using seq_in_image
+    have s'_in_U := fun n ↦ (hs n).1
+    have seq_factor : seq = f ∘ s' := by
+      ext n
+      simp
+      rw [<-(hs n).2]
+    -- the sequence and its limit is compact, so is its preimage by properness
+    set cluster_seq := (insert y (Set.range seq))
+    have preim_comp := h cluster_seq (Tendsto.isCompact_insert_range seq_conv)
+    have s'_im : ∀ n, s' n ∈ (Set.preimage f cluster_seq) := by
+      intro n
+      simp
+      change (f ∘ s') n ∈ cluster_seq
+      rw [<-seq_factor]
+      right
+      simp
+    -- by compactness we have a converging subsequence
+    obtain ⟨a, _, φ, hstrict, htendsto ⟩ := IsCompact.tendsto_subseq preim_comp s'_im
+    -- tedious rewriting
+    have fs'_tendsto := tendsTo_comp_continuous htendsto hcont
+    simp at fs'_tendsto
+    change Tendsto ((f ∘ s') ∘ φ) _ _ at fs'_tendsto
+    rw [<- seq_factor] at fs'_tendsto
+    -- subsequence still converges
+    replace hconv := seq_conv.comp (StrictMono.tendsto_atTop hstrict)
+    have aU : a ∈ U := by
+      -- a closed set is sequencially closed
```

Comment:
There is nothing wrong with using sequential closedness here, but note that the theorem "the limits of a sequence with values in a closed set stay in that closed set" has actually nothing to do with sequences (see [IsClosed.mem_of_tendsto](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Basic.html#IsClosed.mem_of_tendsto)), and I believe you can make your proof a bit shorter by using it directly.